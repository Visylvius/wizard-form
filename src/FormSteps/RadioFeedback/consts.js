const radioOptions = [
  'Very Difficult',
  'Difficult',
  'Not Difficult nor Easy',
  'Easy',
  'Very Easy',
];

const EASE_OF_USE = 'easeOfUse';

export { EASE_OF_USE, radioOptions };
