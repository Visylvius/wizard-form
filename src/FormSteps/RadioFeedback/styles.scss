@import '../../variables';

.RadioFeedback-container {
  display: flex;
  flex-direction: column;

  .RadioFeedback-inputContainer {
    margin: 10px 0 0 15px;
  }

  .RadioFeedback-RadioInput {
    margin-bottom: 15px;
  }

  .RadioFeedback-label {
    color: $font-main;
    font-size: 12px;
  }
}