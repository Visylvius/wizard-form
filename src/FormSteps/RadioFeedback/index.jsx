import React from 'react';

import RadioInput from '../../Components/RadioInput';
import Button from '../../Components/Button';

import { EASE_OF_USE, radioOptions } from './consts';
import './styles.scss';

const RadioFeedback = ({
  currentStep,
  onChange,
  onButtonClick,
  radioValue,
}) => {
  if (currentStep !== 4) {
    return null;
  }

  return (
    <div className='RadioFeedback-container'>
      <label className='RadioFeedback-label'>
        How easy was it to accomplish that goal?
      </label>
      <div className='RadioFeedback-inputContainer'>
        {radioOptions.map((option, index) => {
          return (
            <div key={index}>
              <RadioInput
                data-testid={`RadioFeedback-radio${index}`}
                containerClasses='RadioFeedback-RadioInput'
                checked={option === radioValue}
                shouldShowLabel
                labelText={option}
                name={EASE_OF_USE}
                value={option}
                onChange={onChange}
              />
            </div>
          );
        })}
        <Button
          data-testid='RadioFeedback-Button'
          disabled={!radioValue}
          onClick={onButtonClick}
          shouldCenter
        >
          Next
        </Button>
      </div>
    </div>
  );
};

export default RadioFeedback;
