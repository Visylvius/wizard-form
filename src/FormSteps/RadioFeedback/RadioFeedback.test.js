import React from 'react';
import renderer from 'react-test-renderer';
import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import RadioFeedback from './index';

describe('RadioFeedback (Unit)', () => {
  const mockOnChange = jest.fn();
  const mockOnClick = jest.fn();
  const renderRadioFeedback = props =>
    render(
      <RadioFeedback
        currentStep={4}
        onChange={mockOnChange}
        onButtonClick={mockOnClick}
        {...props}
      />,
    );

  describe('render', () => {
    test('should render properly', () => {
      const tree = renderer.create(
        <RadioFeedback
          currentStep={4}
          onChange={mockOnChange}
          onButtonClick={mockOnClick}
        />,
      );

      expect(tree).toMatchSnapshot();
    });

    test('should render null if the step is not 5', () => {
      const tree = renderer.create(
        <RadioFeedback
          currentStep={2}
          onChange={mockOnChange}
          onButtonClick={mockOnClick}
        />,
      );

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Intergration', () => {
    test('should properly call onChange', async () => {
      const { getByTestId } = renderRadioFeedback();

      userEvent.click(getByTestId('RadioFeedback-radio0'));

      await waitFor(() => expect(mockOnChange).toHaveBeenCalled());
    });

    test('should properly call onClick', async () => {
      const { getByTestId } = renderRadioFeedback({ radioValue: 'Very Easy' });

      userEvent.click(getByTestId('RadioFeedback-Button'));

      await waitFor(() => expect(mockOnClick).toHaveBeenCalled());
    });

    test('should not call onClick when the button is disabled', async () => {
      const { getByTestId } = renderRadioFeedback();

      userEvent.click(getByTestId('RadioFeedback-Button'));

      await waitFor(() => expect(mockOnClick).not.toHaveBeenCalled());
    });
  });
});
