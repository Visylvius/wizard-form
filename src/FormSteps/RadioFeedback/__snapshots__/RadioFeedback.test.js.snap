// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioFeedback (Unit) render should render null if the step is not 5 1`] = `null`;

exports[`RadioFeedback (Unit) render should render properly 1`] = `
<div
  className="RadioFeedback-container"
>
  <label
    className="RadioFeedback-label"
  >
    How easy was it to accomplish that goal?
  </label>
  <div
    className="RadioFeedback-inputContainer"
  >
    <div>
      <div
        className="RadioInput-container RadioFeedback-RadioInput"
      >
        <input
          checked={false}
          className="RadioInput-component"
          data-testid="RadioFeedback-radio0"
          name="easeOfUse"
          onChange={[MockFunction]}
          type="radio"
          value="Very Difficult"
        />
        <label
          className="RadioInput-label"
        >
          Very Difficult
        </label>
      </div>
    </div>
    <div>
      <div
        className="RadioInput-container RadioFeedback-RadioInput"
      >
        <input
          checked={false}
          className="RadioInput-component"
          data-testid="RadioFeedback-radio1"
          name="easeOfUse"
          onChange={[MockFunction]}
          type="radio"
          value="Difficult"
        />
        <label
          className="RadioInput-label"
        >
          Difficult
        </label>
      </div>
    </div>
    <div>
      <div
        className="RadioInput-container RadioFeedback-RadioInput"
      >
        <input
          checked={false}
          className="RadioInput-component"
          data-testid="RadioFeedback-radio2"
          name="easeOfUse"
          onChange={[MockFunction]}
          type="radio"
          value="Not Difficult nor Easy"
        />
        <label
          className="RadioInput-label"
        >
          Not Difficult nor Easy
        </label>
      </div>
    </div>
    <div>
      <div
        className="RadioInput-container RadioFeedback-RadioInput"
      >
        <input
          checked={false}
          className="RadioInput-component"
          data-testid="RadioFeedback-radio3"
          name="easeOfUse"
          onChange={[MockFunction]}
          type="radio"
          value="Easy"
        />
        <label
          className="RadioInput-label"
        >
          Easy
        </label>
      </div>
    </div>
    <div>
      <div
        className="RadioInput-container RadioFeedback-RadioInput"
      >
        <input
          checked={false}
          className="RadioInput-component"
          data-testid="RadioFeedback-radio4"
          name="easeOfUse"
          onChange={[MockFunction]}
          type="radio"
          value="Very Easy"
        />
        <label
          className="RadioInput-label"
        >
          Very Easy
        </label>
      </div>
    </div>
    <div
      className="Button-container"
    >
      <button
        className="Button-component"
        data-testid="RadioFeedback-Button"
        disabled={true}
        onClick={[MockFunction]}
      >
        Next
      </button>
    </div>
  </div>
</div>
`;
